Core/include/Acts/TrackFitting/GaussianSumFitter.hpp

```diff
@@ -440,6 +440,15 @@ struct GaussianSumFitter {
       track.parameters() = params.parameters();
       track.covariance() = params.covariance().value();
       track.setReferenceSurface(params.referenceSurface().getSharedPtr());
+      
+      using MultiPars = std::optional<Acts::MultiComponentBoundTrackParameters<SinglyCharged>>;
+      const std::string key = "multiComponentState";
+      if(not trackContainer.hasColumn(key)) {
+        trackContainer.template addColumn<MultiPars>(key);
+      }
```

Comment:
Now here's a big question: should the tools / fitters add columns, or should they only react to the columns being there, and skip writing the data if they're not?

Reply:
Good question. Probably the "reaction" way is better if we think of application in Athena or something else, because they might want to opt out of this.

Reply:
Makes sense to me.